tools/opt.c

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * Copyright (C) 2011-2013  ProFUSION embedded systems
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#include <shared/strbuf.h>
+
+#include "kmod.h"
+
+int options_from_array(char **args, int nargs, char **output)
+{
+	struct strbuf sbuf;
+	int i;
+
+	if (nargs == 0) {
+		*output = NULL;
+		return 0;
+	}
+
+	strbuf_init(&sbuf);
+	for (i = 0; i < nargs; i++) {
+		bool quote = false;
+		char *value;
+
+		value = strchr(args[i], '=');
+		if (value) {
+			*value++ = '\0';
+			if (*value != '"' && *value != '\'') {
+				if (strchr(value, ' '))
+					quote = true;
+			}
+		}
+
+		if (sbuf.used > 0) {
+			if (!strbuf_pushchar(&sbuf, ' '))
+				goto fail;
+		}
+
+		if (quote) {
+			if (!strbuf_pushchars(&sbuf, args[i]) ||
```

Comment:
Yeah, I have thought about it as well, but for now, it would be its only user. I haven't found another use case yet, so I dropped the idea to not add an unused function into library.

In fact, let's keep the function as it is right now, after all it works (and doesn't modify the input).